e2e/endpointslice: Add Ready Conditions

aauren opened the pull request:
#### What type of PR is this?

/kind bug
/kind cleanup

#### What this PR does / why we need it:

Adds the readiness condition to manually made Endpoints of the EndpointSlice e2e tests so that the readiness of the endpoints is no longer ambiguous and works across CNIs that filter for readiness conditions.

#### Which issue(s) this PR fixes:
Fixes #122623 

#### Special notes for your reviewer:

FYI @aojea @hakman 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

aauren commented:
/sig network
/kind failing-testing

aojea commented:
/ok-to-test
/lgtm
/approve

Thanks

aojea commented:
@aauren you need to remove the reference from the commit message

> Fixes: https://github.com/kubernetes/kubernetes/issues/122623

aauren committed:
e2e/endpointslice: add ready conditions

Add ready conditions to the Endpoints of the self-generated
EndpointSlice tests so that the readiness is not ambiguous and it will
work across CNIs that filter for ready endpoints.

aauren commented:
Sorry about that, I wasn't aware. I've amended the commit and re-pushed. It should be better now.

aojea commented:
you have a gofmt error 

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/122633/pull-kubernetes-verify/1744117291722739712

don't worry about the commit msg error, is a kubernetes CI only thing https://groups.google.com/g/kubernetes-dev/c/MOnwc8_76wo/m/546zDpEsBwAJ

aauren commented:
Sheesh... Rookie mistake. Sorry about that.

I gofmt'ed my file again and verified that `hack/verify-gofmt.sh` completes successfully this time.

hakman commented:
/lgtm
